stan/math/prim/mat.hpp

```diff
@@ -46,6 +46,8 @@
 #include <stan/math/prim/mat/err/constraint_tolerance.hpp>
 #include <stan/math/prim/mat/err/validate_non_negative_index.hpp>
 
+#include <stan/math/prim/mat/vectorize/apply_scalar_unary.hpp>
```

Comment:
It failed to compile when I moved `stan/math/rev/mat/vectorize/apply_scalar_unary.hpp` after `stan/math/prim/mat.hpp` in `stan/math/rev/mat.hpp` . I get the following error messages, which look like the messages that I got earlier when trying to vectorize exp.hpp:

```
In file included from test/unit/math/mix/mat/vectorize/foo_test.cpp:1:
In file included from ./stan/math/mix/mat.hpp:12:
In file included from ./stan/math/prim/mat.hpp:91:
In file included from ./stan/math/prim/mat/fun/exp.hpp:4:
./stan/math/prim/mat/vectorize/apply_scalar_unary.hpp:40:41: error: implicit
      instantiation of undefined template
      'Eigen::internal::traits<stan::math::var>'
      typedef typename Eigen::internal::traits<T>::Scalar scalar_t;
                                        ^
./stan/math/prim/mat/fun/log.hpp:31:21: note: in instantiation of template class
      'stan::math::apply_scalar_unary<stan::math::log_fun, stan::math::var>'
      requested here
    inline typename apply_scalar_unary<log_fun, T>::return_t
                    ^
./stan/math/rev/scal/fun/grad_inc_beta.hpp:26:16: note: while substituting
      deduced template arguments into function template 'log' [with T =
      stan::math::var]
      var c1 = log(z);
               ^
lib/eigen_3.2.8/Eigen/src/Core/util/ForwardDeclarations.h:17:29: note: template
      is declared here
template<typename T> struct traits;
                            ^
In file included from test/unit/math/mix/mat/vectorize/foo_test.cpp:1:
In file included from ./stan/math/mix/mat.hpp:12:
In file included from ./stan/math/prim/mat.hpp:91:
In file included from ./stan/math/prim/mat/fun/exp.hpp:4:
./stan/math/prim/mat/vectorize/apply_scalar_unary.hpp:46:42: error: no member
      named 'RowsAtCompileTime' in 'stan::math::var'
      typedef Eigen::Matrix<scalar_t, T::RowsAtCompileTime,
                                      ~~~^
./stan/math/prim/mat/vectorize/apply_scalar_unary.hpp:40:41: error: implicit
      instantiation of undefined template
      'Eigen::internal::traits<stan::math::var>'
      typedef typename Eigen::internal::traits<T>::Scalar scalar_t;
                                        ^
./stan/math/prim/mat/fun/exp.hpp:31:21: note: in instantiation of template class
      'stan::math::apply_scalar_unary<stan::math::exp_fun, stan::math::var>'
      requested here
    inline typename apply_scalar_unary<exp_fun, T>::return_t
                    ^
./stan/math/rev/scal/fun/grad_inc_beta.hpp:28:16: note: while substituting
      deduced template arguments into function template 'exp' [with T =
      stan::math::var]
      var c3 = exp(lbeta(a, b)) * inc_beta(a, b, z);
               ^
lib/eigen_3.2.8/Eigen/src/Core/util/ForwardDeclarations.h:17:29: note: template
      is declared here
template<typename T> struct traits;
                            ^
In file included from test/unit/math/mix/mat/vectorize/foo_test.cpp:1:
In file included from ./stan/math/mix/mat.hpp:12:
In file included from ./stan/math/prim/mat.hpp:91:
In file included from ./stan/math/prim/mat/fun/exp.hpp:4:
./stan/math/prim/mat/vectorize/apply_scalar_unary.hpp:46:42: error: no member
      named 'RowsAtCompileTime' in 'stan::math::var'
      typedef Eigen::Matrix<scalar_t, T::RowsAtCompileTime,
                                      ~~~^
4 errors generated.
make: *** [test/unit/math/mix/mat/vectorize/foo_test.o] Error 1
make test/unit/math/mix/mat/vectorize/foo_test failed
exit now (04/21/16 14:48:06 EDT)
```